Feature/41 renv rsconnect updates

GuidoMaggio opened the pull request:
Update of renv chapter. See also #41 

- added a phrase about `snapshot.type = "explicit" / "implicit"`
- updated what is generated in the "renv" folder
- Mentioned in an additional sentence the PPM repo
- `options(repos = "https://packagemanager.rstudio.com/cran/2023-01-02")` works to fix the date of the repo, like MRAN was working previously. tested.
- added a second sentence about `renv::snapshot(dev = TRUE)`
- rephrased/improved "Caching the packages' versions", adding new insight from the help file of `renv::status`
- Added few lines and examples about `renv::checkout()`

I have tested the usage of `renv::checkout(date = , repos = )` and `options(repo = )`. Also checkout is then setting the new repo to the date specified (or to the repos url). 
I.e. `renv::checkout(date = "2023-01-01" )` would then set also  `options(repos = "https://packagemanager.rstudio.com/cran/2023-01-01")`.

Checkout has more options, packages can be installed, or only a snapshot of the `renv.lock` can be done.

Update of CI-CD chapter.

- updated workflows yml to mirror ShinyCICD repo
- added `forceUpdate = TRUE` to `rsconnect::deployApp`

riccardoporreca reviewed (changes requested):
@guidomaggio, the workflows are no fully up-to-date with ShinyCICD

When porting updates done and tested in ShinyCICD to the techguides, I would always try to get in ShinyCICD the workflows from the techguides to verify we have full consistency, via

```r
usethis::use_github_file("miraisolutions/techguides", ref="feature/41-renv-rsconnect-updates",path="shiny-ci-cd/actions/ci-cd.yml", save_as = "shiny-ci-cd/actions/ci-cd.yml")
usethis::use_github_file("miraisolutions/techguides", ref="feature/41-renv-rsconnect-updates",path="shiny-ci-cd/actions/ci-cd-renv.yml", save_as = "shiny-ci-cd/actions/ci-cd-renv.yml")
```

Also `deploy-shinyapps.io` lacks some of the refinements we did in ShinyCICD
(NOTE: we cannot use `usethis::use_github_action()`, as we have in the techguides for a branch that contains "/").

GuidoMaggio commented:
@riccardoporreca I have added 'renv' instead of 'release' in the -renv.yml. Here I did not spot other differences with ShinyCICD.
As for the deploy script, I added the appFiles argument.

```
usethis::use_github_file("miraisolutions/techguides", ref="feature/41-renv-rsconnect-updates",path="shiny-ci-cd/actions/ci-cd.yml", save_as = "shiny-ci-cd/actions/ci-cd.yml")
usethis::use_github_file("miraisolutions/techguides", ref="feature/41-renv-rsconnect-updates",path="shiny-ci-cd/actions/ci-cd-renv.yml", save_as = "shiny-ci-cd/actions/ci-cd-renv.yml")
```
The code above works well to me

riccardoporreca commented:
> Here I did not spot other differences with ShinyCICD.
> As for the deploy script, I added the appFiles argument.

There is more in both cases that we want to have. Maybe it is easier for me to take over, since some of the changes should probably be explained in the techguides as part of this content refresh. @GuidoMaggio let's coordinate on this.

Btw, in the past I actually pulled in the full files from ShinyCICD to avoid manual pasting and have more traceability, see https://github.com/miraisolutions/techguides/commit/092de7537470655333d3a2c14f7ae9ce047296cc

riccardoporreca commented:
@GuidoMaggio, I have pushed my updates and general review of the Shiny CI/CD chapter.

I plan to do the same for the renv chapter

riccardoporreca commented:
@GuidoMaggio, this is complete from my side, please cross-review the changes I did

GuidoMaggio reviewed (commented):
2 changes requested

riccardoporreca commented on renv.Rmd:
```diff
@@ -3,95 +3,117 @@
 
-- the creation of a new `renv.lock` file where the R version and the list of used packages with their version are tracked;
+- the creation of the lock-file `renv.lock`, which tracks the version of the resolved transitive package dependencies, along with the R version and package repositories information;
+- the set-up of `renv` infrastructure for the project including a specific library;
```
Correct, I think we can phrase it as follows, since the part about "the generation of the list of packages currently being used" is covered by the generation of `renv.lock`

```suggestion
- the set-up of `renv` infrastructure for the project including a specific library, where packages are installed at the versions tracked in `renv.lock`;
```
